src/components/Carousel.ts

```diff
@@ -334,11 +344,24 @@ export default defineComponent({
       }, config.autoplay);
     }
 
+    function resetAutoplayTimer(restart = true): void {
+      if (!autopayTimer.value) {
+        return
+      }
+
+      clearInterval(autopayTimer.value);
+      if (restart) {
+        initializeAutoplay();
+      }
+    }
+
     /**
      * Navigation function
      */
     const isSliding = ref(false);
     function slideTo(slideIndex: number, mute = false): void {
+      resetAutoplayTimer();
```

Comment:
Reset before `currentSlideIndex.value === slideIndex || isSliding.value` check, to prevent autoplay if user tries paginate to current slide using pagination